[MediaCard] Add border-radius when necessary

alex-page opened the pull request:
### WHY are these changes introduced?

Fixes minor visual bug with VideoThumbnail

### WHAT is this pull request doing?

Adds a top left and right border-radius to the image in the VideoThumbnail
## <!-- ℹ️ Delete the following for small / trivial changes -->

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

Look at the VideoThumbnail example in Storybook

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated the component's `README.md` with documentation changes
* [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [x] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

dfmcphee commented:
This looks good to me, but I think we may need a fix for the image media cards as well.

<img width="600" alt="Screen Shot 2020-04-15 at 10 32 45 AM" src="https://user-images.githubusercontent.com/478990/79349792-ad298e80-7f04-11ea-864b-23db421ffee0.png">

I think adding this to the 'MediaContainer' would do it

```
    border-top-left-radius: border-radius();
    border-top-right-radius: border-radius();
    overflow: hidden;
```

kyledurand commented:
This looks good for the portrait card but not so much on landscape

![image](https://screenshot.click/All_Components__Media_card_-_Video_card__Storybook_2020-04-15_10-46-44.png)

alex-page commented:
I have moved the change to `.MediaContainer` and fixed up the responsive problems.

Video thumbnail desktop
![Screen Shot 2020-04-15 at 11 19 06 AM](https://user-images.githubusercontent.com/19199063/79354964-16ac9b80-7f0b-11ea-9b78-48d7ce7af6b2.png)

Video thumbnail mobile
![Screen Shot 2020-04-15 at 11 19 16 AM](https://user-images.githubusercontent.com/19199063/79354953-13b1ab00-7f0b-11ea-87db-8cf7234d5c45.png)

Media thumbnail portrait mobile
![Screen Shot 2020-04-15 at 11 19 36 AM](https://user-images.githubusercontent.com/19199063/79355041-2d52f280-7f0b-11ea-878b-69c26518271c.png)


Media thumbnail portrait desktop
![Screen Shot 2020-04-15 at 11 19 42 AM](https://user-images.githubusercontent.com/19199063/79355046-2f1cb600-7f0b-11ea-9d82-975aad196943.png)
